### DISCUSSION TITLE
visibleInShowHideMenu for row selection and actions column?

### DISCUSSION THREAD

**Discussion by matttk** [KevinVandy/mantine-react-table#394] [Ideas]
The `visibleInShowHideMenu` option is great for hiding columns from the show/hide columns menu, but there are two options that are permanently there and always disabled: selection and actions (if you have these options enabled).

These aren't useful for the user, as they can never toggle them and would never want to toggle them. The only way I know of to hide these options from the menu is with `visibleInShowHideMenu` in the column definition, but these two columns are automatically generated when their respective options are enabled.

We need the option to hide these options as well from the show/hide menu.

**Comment by KevinVandy** [comment]
Just set enableHiding to true for those columns

**Reply by matttk** [reply]
I don't think that's the correct option. Wouldn't that hide the columns entirely? But I did find `displayColumnDefOptions` and realised this exposes exactly what I want.

```
displayColumnDefOptions: {
  "mrt-row-select": {
    visibleInShowHideMenu: false
  },
  "mrt-row-actions": {
    visibleInShowHideMenu: false
  }
}
```

I completely missed [this section](https://v2.mantine-react-table.com/docs/guides/display-columns#display-(built-in)-columns) in the docs. Sorry and thanks for the hint that pointed me in the right direction!
